blockchain/standalone/pow_test.go

```diff
@@ -353,3 +393,1613 @@ func TestCheckProofOfWork(t *testing.T) {
+			{263, 771983, 0x1d00aabd},
+			{264, 772304, 0x1d00aacc},
+			{265, 775659, 0x1d00b362},
+			{266, 776129, 0x1d00b3e0},
+			{267, 778615, 0x1d00ba4b},
+			{268, 782911, 0x1d00c69d},
+			{269, 785122, 0x1d00ccca},
+			{270, 792634, 0x1d00e5e6},
+			{271, 806060, 0x1d00ffff},
+			{272, 813100, 0x1d00ffff},
+			{273, 820611, 0x1d00ffff},
+			{274, 821642, 0x1d00ffff},
+			{275, 823392, 0x1d00ffff},
+			{276, 824267, 0x1d00ffff},
+			{277, 825093, 0x1d00ffff},
+			{278, 826268, 0x1d00ffff},
+			{279, 826731, 0x1d00ffff},
+			{280, 833358, 0x1d00ffff},
+			{281, 833366, 0x1d00ffff},
+			{282, 835141, 0x1d00ffff},
+			{283, 843181, 0x1d00ffff},
+			{284, 846510, 0x1d00ffff},
+			{285, 848500, 0x1d00ffff},
+			{286, 855266, 0x1d00ffff},
+			{287, 857364, 0x1d00ffff},
+			{288, 857850, 0x1d00ffff},
+		},
+	}, {
+		name:          "mainnet: 300s blocks across signed 32-bit int height",
+		params:        &mockMainNetParams,
+		startDiffBits: 0x1b00a5a6,
+		startHeight:   2147483645,
+		startTime:     7157978,
+		perInstanceTests: []perInstanceTest{
+			{2147483646, 7158278, 0x1b00a5a6},
+			{2147483647, 7158578, 0x1b00a5a6},
+			{2147483648, 7158878, 0x1b00a5a6},
+		},
+	}, {
+		name: "mainnet: 300s blocks across max 64-bit signed int height and " +
+			"32-bit max signed int time",
+		params:        &mockMainNetParams,
+		startDiffBits: 0x1b00a5a6,
+		startHeight:   9223372036854775804,
+		startTime:     2147482747,
+		perInstanceTests: []perInstanceTest{
+			{9223372036854775805, 2147483047, 0x1b00a5a6},
+			{9223372036854775806, 2147483347, 0x1b00a5a6},
+			{9223372036854775807, 2147483647, 0x1b00a5a6},
+			{9223372036854775808, 2147483947, 0x1b00a5a6},
+			{9223372036854775809, 2147484247, 0x1b00a5a6},
+		},
+	}, {
+		name:          "testnet: steady 120s blocks at proof of work limit",
+		params:        &mockTestNetParams,
+		startDiffBits: 0x1e00ffff,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 120, 0x1e00ffff},
+			{2, 240, 0x1e00ffff},
+			{3, 360, 0x1e00ffff},
+			{4, 480, 0x1e00ffff},
+			{5, 600, 0x1e00ffff},
+			{6, 720, 0x1e00ffff},
+			{7, 840, 0x1e00ffff},
+			{8, 960, 0x1e00ffff},
+			{9, 1080, 0x1e00ffff},
+			{10, 1200, 0x1e00ffff},
+		},
+	}, {
+		name: "testnet: random solve times per Poisson distribution for a " +
+			"stable hashrate",
+		params:        &mockTestNetParams,
+		startDiffBits: 0x1c014388,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 3, 0x1c012119},
+			{2, 34, 0x1c010958},
+			{3, 51, 0x1c00f045},
+			{4, 126, 0x1c00e614},
+			{5, 135, 0x1c00cec4},
+			{6, 431, 0x1c00f4f2},
+			{7, 465, 0x1c00e179},
+			{8, 588, 0x1c00e220},
+			{9, 886, 0x1c010c6e},
+			{10, 936, 0x1c00faeb},
+			{11, 1516, 0x1c0186b8},
+			{12, 1599, 0x1c017910},
+			{13, 1829, 0x1c01a322},
+			{14, 1900, 0x1c018fd6},
+			{15, 2005, 0x1c018a1d},
+			{16, 2106, 0x1c0182fb},
+			{17, 2157, 0x1c016a22},
+			{18, 2160, 0x1c014388},
+		},
+	}}
```

Comment:
This should add an additional test to trigger potential internal calculation overflow boundaries in case the arbitrary precision integers are changed out for a fixed precision type in the future.

Also, adding a couple of tests using simnet parameters would be ideal too since the proof of work limit on simnet is 255 bits which means the internal calculations will need more than 256 bits without resorting to contrived values as the aforementioned suggested overflow boundary test would.

Reply:
Both of these have been addressed and are now part of the relevant commit.